.github/workflows/snyk.yml

```diff
@@ -0,0 +1,54 @@
+name: Snyk SBOM CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  snyk_scans:
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - uses: actions/checkout@v3
+      - name: Run Snyk to check for vulnerabilities - sarif output
+        continue-on-error: true
+        uses: snyk/actions/golang@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          command: test
+          args:  --all-projects --sarif-file-output=${{ github.event.repository.name }}.sarif --strict-out-of-sync=false
+
+      - name: Run Snyk to check for vulnerabilities - json output
+        continue-on-error: true
+        uses: snyk/actions/golang@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          command: test
+          args: --json --all-projects --json-file-output=${{ github.event.repository.name }}.json  --strict-out-of-sync=false
+
+      - name: install snyk-to-html
```

Comment:
i suppose a similar question here as before. any particular reason to generate all these different formats?

Reply:
Per customer requests mostly. Some customers want the SARIF format, some SPDX, some HTML reports.